packages/presentation/lib/src/pages/splash_screen/bloc/splash_screen_bloc.dart

```diff
@@ -1,27 +1,73 @@
 import 'package:domain/domain.dart';
+import 'package:presentation/generated/l10n.dart';
 import 'package:presentation/src/base_bloc/bloc.dart';
+import 'package:presentation/src/navigation/base_arguments.dart';
 import 'package:presentation/src/pages/home_page/home.dart';
+import 'package:presentation/src/pages/splash_screen/bloc/alert_dialog_data.dart';
 
-abstract class SplashScreenBloc extends Bloc {
+abstract class SplashScreenBloc extends Bloc<BaseArguments, AlertDialogData?> {
   factory SplashScreenBloc(
-    InitialApiCallUseCase initialApiCall,
+    CheckAppVersionUseCase checkVersion,
+    UpdateAppUseCase updateApp,
   ) =>
-      _SplashScreenBloc(initialApiCall);
+      _SplashScreenBloc(checkVersion, updateApp);
 }
 
-class _SplashScreenBloc extends BlocImpl implements SplashScreenBloc {
-  final InitialApiCallUseCase initialApiCall;
+class _SplashScreenBloc extends BlocImpl<BaseArguments, AlertDialogData?>
+    implements SplashScreenBloc {
+  final CheckAppVersionUseCase checkVersion;
+  final UpdateAppUseCase updateApp;
 
-  _SplashScreenBloc(this.initialApiCall) : super(null);
+  _SplashScreenBloc(
```

Comment:
I will point it out) according to the assignment, we should not remove our delay, but return to it

Reply:
we already discussed in your pr =)